pkg/buildkit/buildkit.go

```diff
@@ -120,27 +76,78 @@ func InitializeBuildkitConfig(ctx context.Context, client *client.Client, image
 	}
 
 	// Resolve and pull the config for the target image
-	_, configData, err := resolveImageConfig(ctx, image, &config.Platform)
+	_, _, configData, err := c.ResolveImageConfig(ctx, image, llb.ResolveImageConfigOpt{
+		ResolveMode: llb.ResolveModePreferLocal.String(),
+	})
 	if err != nil {
 		return nil, err
 	}
+
 	config.ConfigData = configData
 
 	// Load the target image state with the resolved image config in case environment variable settings
 	// are necessary for running apps in the target image for updates
 	config.ImageState, err = llb.Image(image,
 		llb.Platform(config.Platform),
-		llb.ResolveModeDefault,
+		llb.ResolveModePreferLocal,
+		llb.WithMetaResolver(c),
 	).WithImageConfig(config.ConfigData)
 	if err != nil {
 		return nil, err
 	}
 
-	config.Client = client
+	config.Client = c
 
 	return &config, nil
 }
 
+// Extracts the bytes of the file denoted by `path` from the state `st`.
+func ExtractFileFromState(ctx context.Context, c gwclient.Client, st *llb.State, path string) ([]byte, error) {
+	def, err := st.Marshal(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Solve(ctx, gwclient.SolveRequest{
+		Evaluate:   true,
+		Definition: def.ToPB(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ref, err := resp.SingleRef()
+	if err != nil {
+		return nil, err
+	}
+
+	return ref.ReadFile(ctx, gwclient.ReadRequest{
+		Filename: path,
+	})
+}
+
+func ArrayFile(input []string) []byte {
```

Comment:
should be easy to add a unit test for this?

Reply:
- [ ] Unit Test